src/sentry/workflow_engine/processors/detector.py

```diff
@@ -97,6 +114,107 @@ def _ensure_detector(project: Project, type: str) -> Detector:
+        Detector.objects.filter(type=MetricIssue.slug, project=project).order_by("id").first()
+    )
+    if existing:
+        return existing
+
+    lock = locks.get(
+        f"workflow-engine-project-{MetricIssue.slug}-detector:{project.id}",
+        duration=2,
+        name=f"workflow_engine_default_{MetricIssue.slug}_detector",
+    )
+    try:
+        with (
+            lock.blocking_acquire(initial_delay=0.1, timeout=3),
+            transaction.atomic(router.db_for_write(Detector)),
+        ):
+            # Double-check after acquiring lock in case another process created it
+            existing = (
+                Detector.objects.filter(type=MetricIssue.slug, project=project)
+                .order_by("id")
+                .first()
+            )
+            if existing:
+                return existing
+
+            condition_group = DataConditionGroup.objects.create(
+                logic_type=DataConditionGroup.Type.ANY,
+                organization_id=project.organization_id,
+            )
+
+            DataCondition.objects.create(
+                comparison={
+                    "sensitivity": AnomalyDetectionSensitivity.LOW,
+                    "seasonality": AnomalyDetectionSeasonality.AUTO,
+                    "threshold_type": AnomalyDetectionThresholdType.ABOVE,
+                },
+                condition_result=DetectorPriorityLevel.HIGH,
+                type=Condition.ANOMALY_DETECTION,
+                condition_group=condition_group,
+            )
+
+            detector = Detector.objects.create(
+                project=project,
+                name="High Error Count (Default)",
+                description="Automatically monitors for anomalous spikes in error count",
+                workflow_condition_group=condition_group,
+                type=MetricIssue.slug,
+                config={
+                    "detection_type": AlertRuleDetectionType.DYNAMIC.value,
+                    "comparison_delta": None,
+                },
+                owner_team_id=owner_team_id,
+                enabled=enabled,
+            )
+
+            snuba_query = create_snuba_query(
+                query_type=SnubaQuery.Type.ERROR,
+                dataset=Dataset.Events,
+                query="",  # filter????
+                aggregate="count()",
+                time_window=timedelta(minutes=15),
+                resolution=timedelta(minutes=1),
+                environment=None,
+                event_types=[SnubaQueryEventType.EventType.ERROR],
+            )
+
+            query_subscription = create_snuba_subscription(
+                project=project,
+                subscription_type=INCIDENTS_SNUBA_SUBSCRIPTION_TYPE,
+                snuba_query=snuba_query,
+            )
+
+            data_source = DataSource.objects.create(
+                organization_id=project.organization_id,
+                source_id=str(query_subscription.id),
+                type="snuba_query_subscription",
```

Comment:
Can you use `DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION` from https://github.com/getsentry/sentry/blob/master/src/sentry/incidents/utils/types.py#L45?